#! /usr/bin/env python

PACKAGE = "rqt_quad_control_tuning"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

trajectory_enum = gen.enum([    
                gen.const("none",       int_t,  0,  "No command"),
                gen.const("horizontal_circle",         int_t,  1,  "Horizontal circular trajectory"),
                gen.const("vertical_circle",         int_t,  2,  "Vertical circular trajectory"),
                gen.const("minimum_snap", int_t, 3, "Minimum Snap Waypoints Trajectory"),
                gen.const("lemniscate",       int_t,  4,  "Lemniscate (Figure-8) Trajectory"),
                gen.const("square",         int_t,  5,  "Square trajectoty"),
                gen.const("random_min_snap", int_t, 6, "Continuous Random Minimum snap"),
                gen.const("exit_random_min_snap", int_t, 7, "Exit Continuous Random Minimum snap")
                ], "Trajectory type")

#           Name        Type        Level   Description         Default     Min Max     Values
gen.add(    "trajectory",   int_t,      0,  "Trajectory type",      0,      0,  12, edit_method=trajectory_enum)
gen.add(    "speed",    double_t,   1,      "Trajectory speed (m/s)",   1,      0,  5.0)
gen.add(    "circle_radius",    double_t,       2,      "Circle trajectory radius",  1.0,      0,  3.0)
gen.add(    "circle_orientation",    double_t,       2,      "Vertical circle orientation",  0.0,      0,  6.28)
gen.add(    "num_rotations",    double_t,       2,      "Number of rotations",  1,      0.5,  20)
gen.add(    "max_thrust",    double_t,       2,      "Maximum thrust",  15,      1,  30)
gen.add(    "max_roll_pitch_rate",    double_t,       2,      "Maximum roll/pitch rate",  0.5,      0,  2)

exit(gen.generate(PACKAGE, "rqt_quad_control_tuning", "setTrajectory"))